redfish/provider/resource_redfish_manager_reset.go

```diff
@@ -0,0 +1,234 @@
+	if req.ProviderData == nil {
+		return
+	}
+	r.p = req.ProviderData.(*redfishProvider)
+	tflog.Trace(ctx, "resource_manager_reset configured")
+}
+
+// Metadata returns the resource type name.
+func (*managerResetResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "manager_reset"
+}
+
+// ManagerResetSchema to design the schema for manager reset resource.
+func ManagerResetSchema() map[string]schema.Attribute {
+	return map[string]schema.Attribute{
+		"id": schema.StringAttribute{
+			MarkdownDescription: "ID of the manager reset resource",
+			Description:         "ID of the manager reset resource",
+			Computed:            true,
+		},
+		"manager_id": schema.StringAttribute{
+			MarkdownDescription: "The value of the Id property of the Manager resource",
+			Description:         "The value of the Id property of the Manager resource",
+			Required:            true,
+			Validators: []validator.String{
+				stringvalidator.LengthAtLeast(1),
+			},
+		},
+		"reset_type": schema.StringAttribute{
+			MarkdownDescription: "The type of ResetType option to be performed",
+			Description:         "The type of ResetType option to be performed",
+			Required:            true,
+			Validators: []validator.String{
+				stringvalidator.OneOf(
+					string(redfish.GracefulRestartResetType),
+				),
+			},
+		},
+	}
+}
+
+// Schema defines the schema for the resource.
+func (*managerResetResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "Resource for managing manager reset.",
+
+		Attributes: ManagerResetSchema(),
+		Blocks:     RedfishServerResourceBlockMap(),
+	}
+}
+
+// Create creates the resource and sets the initial Terraform state.
+func (r *managerResetResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	tflog.Trace(ctx, "resource_manager_reset create : Started")
+	// Get Plan Data
+	var plan models.RedfishManagerReset
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Lock the mutex to avoid race conditions with other resources
+	redfishMutexKV.Lock(plan.RedfishServer[0].Endpoint.ValueString())
+	defer redfishMutexKV.Unlock(plan.RedfishServer[0].Endpoint.ValueString())
+
+	resetType := plan.ResetType.ValueString()
+	managerID := plan.ManagerID.ValueString()
+
+	// Get manager
+	manager, err := getManager(r, plan, managerID)
+	if err != nil {
+		resp.Diagnostics.AddError("Error while retrieving manager from redfish API", err.Error())
+		return
+	}
+
+	// Perform manager reset
+	err = manager.Reset(redfish.ResetType(resetType))
+	if err != nil {
+		resp.Diagnostics.AddError("Error resetting manager", err.Error())
+		return
+	}
+
+	plan.Id = types.StringValue(manager.ODataID)
+
+	// time to allow rebooting iDRAC
+	time.Sleep(300 * time.Second)
```

Comment:
Why do we need to wait for the reboot to be complete inside the provider? Since we are not making an API call after the sleep time, I am confused abt the reason for the sleep.

Reply:
we are waiting here for iDRAC to get rebooted.